lib/src/lib/directives/form-field.directive.ts

```diff
@@ -22,34 +22,42 @@ import { FormFieldMappingService } from '../services/form-field-mapping.service'
   selector: '[lab900FormField]',
   standalone: true,
 })
-export class FormFieldDirective implements OnChanges, OnInit, OnDestroy {
+export class FormFieldDirective implements OnChanges, OnInit {
+  private readonly container = inject(ViewContainerRef);
+  private readonly formFieldMappingService = inject(FormFieldMappingService);
+
   @Input()
   public schema: Lab900FormField;
 
   @Input()
   public group: UntypedFormGroup;
 
-  @Input()
-  public language?: string;
-
-  @Input()
-  public availableLanguages?: ValueLabel[];
+  public readonly language = input<string | undefined>(undefined);
+  public readonly availableLanguages = input<ValueLabel[]>([]);
 
   @Input()
   public readonly = false;
 
   @Input()
   public externalForms?: Record<string, UntypedFormGroup>;
 
-  public component: ComponentRef<FormComponent>;
+  public component?: ComponentRef<FormComponent>;
 
-  public statusChangeSubscription: Subscription;
-
-  public constructor(
-    private resolver: ComponentFactoryResolver,
-    private container: ViewContainerRef,
-    private formFieldMappingService: FormFieldMappingService,
-  ) {}
+  public constructor() {
+    effect(() => {
+      if (this.component) {
```

Comment:
if the `component` is not defined yet, the if check will not run, and the effect will never run again. are you sure this works?
extract `this.availableLanguages()` out of the if check